pkg/replica/diff_disk.go

```diff
@@ -78,16 +78,18 @@ func (d *diffDisk) WriteAt(buf []byte, offset int64) (int, error) {
 		return d.readModifyWrite(buf, offset)
 	}
 
-	if _, err := d.readModifyWrite(buf[0:startCut], offset); err != nil {
-		return 0, err
+	if n, err := d.readModifyWrite(buf[0:startCut], offset); err != nil {
+		return n, err
 	}
 
-	if _, err := d.fullWriteAt(buf[startCut:int64(len(buf))-endOffset], offset+startCut); err != nil {
-		return 0, err
+	startCutWritebytes := len(buf[0:startCut])
+	if n, err := d.fullWriteAt(buf[startCut:int64(len(buf))-endOffset], offset+startCut); err != nil {
+		return n + startCutWritebytes, err
 	}
 
-	if _, err := d.readModifyWrite(buf[int64(len(buf))-endOffset:], offset+int64(len(buf))-endOffset); err != nil {
-		return 0, err
+	fullWriteBytes := len(buf[startCut : int64(len(buf))-endOffset])
```

Comment:
Can't we get `fullWriteBytes` from the returned `n`?

Reply:
Yes, we should use the returned written bytes `n`.
Modified. Thanks.